### PR TITLE
Add rspec & sign in/sign up flow

### PR DESCRIPTION
**Pull request by smurfik**

### CODE REVIEW

File: app/views/users/registrations/edit.html.erb [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  <%= devise_error_messages! %>
+
+  <div class="field">
+    <%= f.label :first_name %><br />
+    <%= f.text_field :first_name, autofocus: true %>
+  </div>
+
+  <div class="field">
+    <%= f.label :last_name %><br />
+    <%= f.text_field :last_name, autofocus: true %>
```
Comment (adambray) [inline app/views/users/registrations/edit.html.erb]: May only want to include the autofocus attribute on the first field of each form

**Comment by adambray** [comment]
I don't see a link to sign in, but we can maybe add that as part of the styling work so we can get this merged in.

File: Gemfile [hunk 2]
```diff
@@ -32,6 +32,8 @@ gem 'jbuilder', '~> 2.5'
 # Use ActiveModel has_secure_password
 # gem 'bcrypt', '~> 3.1.7'
 
+gem 'devise'
```
Comment (keithrbennett) [inline Gemfile]: Do we want a version constraint here?

File: db/migrate/20170519140150_devise_create_users.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      t.string :first_name,        null: false
+      t.string :last_name,         null: false
+      t.string :address,           null: false
+      t.string :city,              null: false
+      t.string :state,             null: false
+      t.string :zipcode,           null: false
+
+      ## Database authenticatable
+      t.string :email,              null: false, default: ""
+      t.string :encrypted_password, null: false, default: ""
```
Comment (keithrbennett) [inline db/migrate/20170519140150_devise_create_users.rb]: Do we want all these fields to be required?

Comment (smurfik) [inline db/migrate/20170519140150_devise_create_users.rb]: yes

File: Gemfile (hunk 2 above)
Comment (smurfik) [inline Gemfile]: great catch! we added the version constraint

**Pull request merged**